dao-macros/src/serialization.rs

```diff
@@ -110,3 +108,86 @@ pub fn derive_to_bytes(input: TokenStream) -> TokenStream {
 
     TokenStream::from(expanded)
 }
+
+fn variants(input: DeriveInput) -> Result<Vec<Ident2>, TokenStream> {
+    let variants = match input.data {
+        Data::Enum(DataEnum { variants, .. }) => {
+            variants.into_iter().map(|x| x.ident).collect::<Vec<_>>()
+        }
+        _ => {
+            return Err(TokenStream::from(
+                quote! { compile_error!("Expected an enum."); },
+            ))
+        }
+    };
+    Ok(variants)
+}
+
+pub fn derive_to_bytes_enum(input: DeriveInput) -> TokenStream {
+    let enum_ident = input.ident.clone();
+    let variants = match variants(input) {
+        Ok(variants) => variants,
+        Err(error_stream) => return error_stream,
+    };
+
+    let mut append_bytes = TokenStream2::new();
+    let mut index = 0;
+    append_bytes.append_all(variants.iter().map(|ident| {
+        index += 1;
+        quote! {
+          #enum_ident::#ident => #index,
+        }
+    }));
```

Comment:
mapping with side effects looks suspicious. I think you can replace it with `enumetare()` to have an index.
```suggestion
append_bytes.append_all(variants.iter().enumerate().map(|(index, ident)| {
        let index = index + 1;
        quote! {
          #enum_ident::#ident => #index,
        }
    }));
```